common/coin_support/solana.h

```diff
@@ -0,0 +1,114 @@
+/**
+ * @file    solana.h
+ * @author  Cypherock X1 Team
+ * @brief   Header for solana coin.
+ *          Stores declarations for solana coin functions.
+ * @copyright Copyright (c) 2022 HODL TECH PTE LTD
+ * <br/> You may obtain a copy of license at <a href="https://mitcc.org/" target=_blank>https://mitcc.org/</a>
+ *
+ */
+#ifndef SOLANA_HEADER
+#define SOLANA_HEADER
+
+#include <stdint.h>
+#include "coin_utils.h"
+
+// Derivation path reference : https://docs.solana.com/wallet-guide/paper-wallet#hierarchical-derivation
+#define SOLANA_PURPOSE_INDEX 0x8000002C
+#define SOLANA_COIN_INDEX    0x800001F5
+#define SOLANA_ACCOUNT_INDEX 0x80000000
+#define SOLANA_CHAIN_INDEX   0x80000000
+#define SOLANA_ADDRESS_INDEX 0x80000001
+
+#define SOLANA_ACCOUNT_ADDRESS_LENGTH 32
+#define SOLANA_BLOCKHASH_LENGTH       32
+
+// Reference : https://docs.rs/solana-program/1.14.3/solana_program/system_instruction/enum.SystemInstruction.html#variant.Transfer
+typedef struct solana_transfer_data {
+  uint8_t *funding_account;
+  uint8_t *recipient_account;
+  uint64_t lamports;
+} solana_transfer_data;
+
+// Reference : https://docs.solana.com/developing/programming-model/transactions#instruction-format
+typedef struct solana_instruction {
+  uint8_t program_id_index;
+  uint16_t account_addresses_index_count;
+  uint8_t *account_addresses_index;
+  uint16_t opaque_data_length;
+  uint8_t *opaque_data;
+  union {
+    solana_transfer_data transfer;
+  } program;
```

Comment:
There is duplicity among the members. the program should be the container for the two members namely, array of account indices and opaque data.

Reply:
basically, `account_addresses_index` has been redefined as `solana_transfer_data.funding_account` & `solana_transfer_data.recipient_account` in some sense (if not directly).

Reply:
this does not seem useful. having the structs dereferenced seems more relevant.